src/engine.cpp

```diff
@@ -508,14 +508,23 @@ bool Engine::IsVariantHidden(CompanyID c) const
  */
 void EngineOverrideManager::ResetToDefaultMapping()
 {
-	this->mappings.clear();
+	EngineID id = 0;
 	for (VehicleType type = VEH_TRAIN; type <= VEH_AIRCRAFT; type++) {
-		for (uint internal_id = 0; internal_id < _engine_counts[type]; internal_id++) {
-			this->mappings.emplace_back(INVALID_GRFID, internal_id, type, internal_id);
+		auto &map = this->mappings[type];
+		map.clear();
+		for (uint internal_id = 0; internal_id < _engine_counts[type]; internal_id++, id++) {
+			map.emplace_back(INVALID_GRFID, internal_id, type, internal_id, id);
 		}
 	}
 }
 
+struct EngineIDMappingComparator {
```

Comment:
This has become a `Projection` instead of a `Comparator`. I am assuming you went from `std::lower_bound` to `std::ranges::lower_bound` at some point.